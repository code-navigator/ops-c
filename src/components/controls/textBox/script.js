export default {
  props: [
    'label',
    'value'
  ],

  computed: {
    input: {
      get () {
        return this.value
      },
      set (newValue) {
        this.$emit('keyup', newValue.toUpperCase())
      }
    }
  }
}
