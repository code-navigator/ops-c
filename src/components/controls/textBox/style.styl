.textBox
  height: 55px

.textBox .v-input__slot::before
  height: 0px