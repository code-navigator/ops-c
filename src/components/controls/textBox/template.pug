v-flex(xs12)
  v-text-field(
    :label="label"
    class="textBox"
    placeholder
    v-model="input"
  )