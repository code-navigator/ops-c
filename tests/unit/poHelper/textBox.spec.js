import { createLocalVue, mount } from '@vue/test-utils'
import { TestHelpers } from './../testHelpers'
import textBox from './../../../src/components/controls/textBox'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuetify)
localVue.use(Vuex)

describe('test.vue', () => {
  let h
  let store
  let wrapper
  const label = 'some label'
  const value = 'some text'

  beforeEach(() => {
    store = new Vuex.Store({})
    wrapper = mount(textBox, {
      localVue,
      store,
      propsData: {
        label,
        value
      }
    })
    h = new TestHelpers(wrapper, expect)
    jest.resetModules()
    jest.clearAllMocks()
  })

  it('should mount without errors', () => {
    h.isInstance()
  })

  it('should display a label', () => {
    h.hasText(label)
  })

  it('should display the text', () => {
    h.hasInputValue(value)
    h.setText('input', 'new text')
    h.hasInputValue('new text')
  })

  it('should change emit a keyup event when the input changes', () => {
    h.trigger('input', 'input')
    h.emits('keyup')
  })

  it('should render correctly', () => {
    h.renders()
  })
})
