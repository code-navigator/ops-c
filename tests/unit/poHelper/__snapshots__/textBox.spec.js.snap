// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test.vue should render correctly 1`] = `
<div class="flex xs12">
  <div class="v-input textBox v-text-field v-input--is-label-active v-input--is-dirty">
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-text-field__slot">
          <label aria-hidden="true" class="v-label v-label--active" style="left: 0px; position: absolute;">some label</label>
          <input aria-label="some label" type="text" placeholder="placeholder">
        </div>
      </div>
      <div class="v-text-field__details">
        <div class="v-messages">
          <div class="v-messages__wrapper"></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
